src/main/java/com/direwolf20/buildinggadgets/common/schema/MutableBlockData.java

```diff
@@ -0,0 +1,67 @@
+package com.direwolf20.buildinggadgets.common.schema;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.math.BlockPos;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public final class MutableBlockData {
+    private BlockPos pos;
+    private BlockState state;
+    @Nullable
+    private CompoundNBT tileNbt;
+
+    public MutableBlockData(BlockPos pos, BlockState state, @Nullable CompoundNBT tileNbt) {
+        setInformation(pos, state, tileNbt);
+    }
+
+    public BlockPos getPos() {
+        return pos;
+    }
+
+    public BlockState getState() {
+        return state;
+    }
+
+    @Nullable
+    public CompoundNBT getTileNbt() {
+        return tileNbt;
+    }
+
+    public MutableBlockData setInformation(BlockPos pos, BlockState state, @Nullable CompoundNBT tileNbt) {
+        this.pos = Objects.requireNonNull(pos, "Cannot have BlockData with null position");
```

Comment:
If there is no @Nullable on the param then the IDE will tell us if we're about to feed null to the method. We don't need to protect against it in this method.

Reply:
@MichaelHillcox that is not correct. IntelliJ will warn you with an `@NotNull` iff you are feeding it with a value that is marked with `@Nullable`. It will nor warn you about unmarked variables being passed through. If you want to go theough the rewrite and consistently mark everything that may be null with `@Nullable` und use NonNullByDefault for everything else, I'd be really happy. Until then this is going to stay.